iron-overlay-manager.html

```diff
@@ -47,41 +43,23 @@
     this.deepActiveElement = null;
     Object.defineProperty(this, 'deepActiveElement', {
       get: function() {
-        var active = document.activeElement;
         // document.activeElement can be null
         // https://developer.mozilla.org/en-US/docs/Web/API/Document/activeElement
-        while (active && active.root && Polymer.dom(active.root).activeElement) {
+        // In case of null, default it to document.body.
+        var active = document.activeElement || document.body;
+        while (active.root && Polymer.dom(active.root).activeElement) {
           active = Polymer.dom(active.root).activeElement;
         }
         return active;
       }.bind(this)
     });
-  };
-
-  /**
-   * If a node is contained in an overlay.
-   * @private
-   * @param {Node} node
-   * @returns {Boolean}
-   */
-  Polymer.IronOverlayManagerClass.prototype._isChildOfOverlay = function(node) {
-    while (node && node !== document.body) {
-      // Use logical parentNode, or native ShadowRoot host.
-      node = Polymer.dom(node).parentNode || node.host;
-      // Check if it is an overlay.
-      if (node && node.behaviors && node.behaviors.indexOf(Polymer.IronOverlayBehaviorImpl) !== -1) {
-        return true;
-      }
-    }
-    return false;
-  };
-
-  Polymer.IronOverlayManagerClass.prototype._applyOverlayZ = function(overlay, aboveZ) {
-    this._setZ(overlay, aboveZ + 2);
-  };
 
-  Polymer.IronOverlayManagerClass.prototype._setZ = function(element, z) {
-    element.style.zIndex = z;
+    // Enable document-wide tap recognizer.
+    Polymer.Gestures.add(document, 'tap', null);
+    // We should be using only 'tap', but this would be a breaking change.
+    var tapEvent = ('ontouchstart' in window) ? 'tap' : 'click';
+    document.addEventListener(tapEvent, this._onCaptureClick.bind(this), true);
```

Comment:
Since we absolutely need to capture the event before any other element, we need to support `click` for non-touch devices. Without this, whoever does `<button onclick="dialog.open()">` would not see its dialog being opened (because button's `click` would fire before document's `tap` http://jsbin.com/laxoda/2/edit?html,output)